fix(integrations): Ensure retryCount is sent as a number

MrKalyanKing opened the pull request:
- [X] I have read the Contributing Guide
- [X] I have signed the CLA
- [X] Lint and unit tests pass locally with my changes
- [X] I have tested my fix locally and it is effective
- [X] I have added necessary documentation (if applicable)

## Proposed changes

This PR fixes a bug where the "Retry Count" field on the Outgoing Webhook form was sending its value as a **string** instead of a **number**. This caused the API to reject the save/update with a `[invalid-params]` error.

I have updated the `outgoingwebhookform.tsx` component. The `TextInput` for `retryCount` now uses an `onChange` handler to capture the event and set the form's state using `e.currentTarget.valueAsNumber`. This ensures the value is always a **number** (or `undefined` if the field is empty), which the API expects.

I was able to reproduce the bug and confirm the fix on my local dev server, which now saves the integration successfully.

## Issue(s)

Closes #373

## Steps to test or reproduce

1.  Go to `Administration` > `Integrations`.
2.  Click `New` and select `Outgoing`.
3.  Fill in the required fields (Name, Channel, URL).
4.  Scroll down to the **Retry Count** field and change the value (e.g., from `6` to `5`).
5.  Click `Save`.
6.  **Before this fix:** An error "must be number" would appear.
7.  **After this fix:** The integration saves successfully with a "Integration has been added" message.

## Issue(s)

Closes #37346
Related to #37356

## Further comments

This is my first contribution. I'm happy to make any changes if needed!

[COMM-77](https://rocketchat.atlassian.net/browse/COMM-77)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Refactor**
  * Outgoing webhook retry-count input updated to a numeric input control with non‑negative step increments, improving input constraints while preserving labels, hints, and accessibility.
* **Bug Fixes**
  * Retry count is now reliably treated and submitted as a numeric value, preventing invalid/non‑numeric entries.
* **Chores**
  * Added release metadata for a patch update.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

[COMM-77]: https://rocketchat.atlassian.net/browse/COMM-77?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MartinSchoeler commented on apps/meteor/client/views/admin/integrations/outgoing/OutgoingWebhookForm.tsx:
```diff
@@ -476,7 +476,27 @@ const OutgoingWebhookForm = () => {
+											type='number'
+											min='0'
+											step='1'
+											// This 'onChange' is the new magic
```
Please don't leave comments on the component, usually we only use comments to explain code that might be counter intuitive or confusing

MrKalyanKing commented:
Hi @MartinSchoeler, thank you for the detailed feedback! This was very helpful.

I've made all the changes you requested:

Switched from TextInput to NumberInput.

Removed the custom onChange and value props to use the RHF defaults correctly.

Removed the extra comments.

Updated the changeset file description.

I've pushed the amended commit. This should be ready for another look. Thanks!
